shared_model/backend/protobuf/transaction_responses/proto_tx_response.hpp

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_PROTO_TX_RESPONSE_HPP
+#define IROHA_PROTO_TX_RESPONSE_HPP
+
+#include "endpoint.pb.h"
+#include "interfaces/transaction_responses/committed_tx_response.hpp"
+#include "interfaces/transaction_responses/stateful_failed_tx_response.hpp"
+#include "interfaces/transaction_responses/stateful_valid_tx_response.hpp"
+#include "interfaces/transaction_responses/stateless_failed_tx_response.hpp"
+#include "interfaces/transaction_responses/stateless_valid_tx_response.hpp"
+#include "interfaces/transaction_responses/tx_response.hpp"
+#include "interfaces/transaction_responses/unknown_tx_response.hpp"
+#include "utils/lazy_initializer.hpp"
+#include "utils/reference_holder.hpp"
+#include "utils/variant_deserializer.hpp"
+
+template <typename... T>
+auto load(const iroha::protocol::ToriiResponse &ar) {
+  unsigned which = ar.tx_status();
+  constexpr unsigned last = boost::mpl::size<T...>::type::value - 1;
```

Comment:
Use the following approach in case number in proto are not consecutive: https://github.com/hyperledger/iroha/blob/f624783484353ff1c8cc19e7dc4e14359d49443b/shared_model/backend/protobuf/commands/proto_command.hpp#L29

Reply:
May you explain why is this better?

Reply:
I mean it's fine for messages with `oneof`, but for enum looks redundant